data/src/main/scala/ackcord/data/message.scala

```diff
@@ -441,7 +448,9 @@ case class GuildGatewayMessage(
     activity: Option[MessageActivity],
     application: Option[MessageApplication],
     messageReference: Option[MessageReference],
-    flags: Option[MessageFlags]
+    flags: Option[MessageFlags],
+    stickers: Option[List[Sticker]],
+    referencedMessage: Option[RawMessage] // TODO: Is this supposed to be RawMessage or GuildGatewayMessage
```

Comment:
Unless you know that messages with references gotten with `REST` request also contain `GuildGatewayMessage`, then leave it as `Message`

Reply:
Yeah, I checked with advaith and he basically works for discord.